src/org/thoughtcrime/securesms/BaseActionBarActivity.java

```diff
@@ -39,13 +39,21 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     protected void onStart() {
-        super.onStart();
         if (ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp() && !isWaitingForResult) {
-            ScreenLockUtil.applyScreenLock(this);
+          ScreenLockUtil.applyScreenLock(this);
+        } else {
+          findViewById(android.R.id.content).setVisibility(View.VISIBLE);
```

Comment:
this does not alway help to hide the content in a reliable way, eg. on my lineage/oreo it does only work sometimes.

also, i fear, this could have side-effects, esp. as the state-change is always done, also when the screen lock is not used.

i would suggest to skip the hiding therefore. with the screen securiy, we have a more reliable option that seems to work better; we should just add a note to recommend the user to enable it.

Reply:
Does screen security on without the hiding is enough? As far as I remember I was able to see content if no hiding was applied, but I could be wrong. If screen security on is enough I would totally agree to drop the hiding part.

Reply:
@r10s in my test it works most of the time when the UI hiding is removed. I saw content one or two times when performing actions pretty fast. How do we want to proceed, should I remove the hiding anyway and we continue testing "in the wild"?

Reply:
> Does screen security on without the hiding is enough?

can you check this out again on your device?

if hiding is still necessary, however, i would do it only if lock-pattern is really enabled, not always.

Reply:
I did, my last comment form an hour ago was referring to that. The "hide only if lock is enabled part" makes sense, I think that's a good trade.

Reply:
/me did not refresh the page before replying :)

Reply:
> The "hide only if lock is enabled part" makes sense, I think that's a good trade.

cool, i think this is a good way to go.

Reply:
Nice, I will wrap the onStop + setVisibility(View.GONE) part, but not the setVisibility(View.VISIBLE) part to avoid problems when enabling / disabling the function via the settings. Calling setVisibility(View.VISIBLE) shouldn't be a problem as it would not change anything if the view is already visible.

Reply:
> Nice, I will wrap the onStop + setVisibility(View.GONE) part, but not the setVisibility(View.VISIBLE) part to avoid problems when enabling / disabling the function via the settings. Calling setVisibility(View.VISIBLE) shouldn't be a problem as it would not change anything if the view is already visible.

Done @r10s